Added new block "topsection".

petrovsky747 opened the pull request:
Images in all blocks are added as a sprite.

petrovsky747 committed:
Added new block "topsection".
Images in all blocks are added as a sprite.

5tronciy commented on styles/styles.css:
```diff
@@ -39,6 +40,199 @@ body {
+    gap: 24px;
+    justify-content: space-around;
+    align-items: center;
+    padding-left: 550px;
```
Such huge paddings must kill the flexibility. Try to view your layout on the viewport with width 1000px and less.

5tronciy commented on styles/styles.css:
```diff
@@ -39,6 +40,199 @@ body {
+
+.topsection__header__items {
+    display: flex;
+    justify-content: space-between;
```
Doesn't work here.

5tronciy commented on styles/styles.css:
```diff
@@ -39,6 +40,199 @@ body {
+    background-position: center;
+    max-width: 1440px;
+    padding: 0;
+    /* height: 100vh; */
```
Is it the useless comment?

5tronciy commented on index.html:
```diff
@@ -10,24 +10,59 @@
+        <label class="input__label topsection__input__label" for="guests"></label>
+        <div class="topsection__inputs">
+          <input class="input topsection__city" id="city" type="text" name="city" placeholder="New York"/>
+          <input class="input topsection__date" id="date" type="date" name="date" placeholder="Tue 15 Sept — Sat 19 Sept"/>
```
Attribute `placeholder` is not allowed with the input type="date".

5tronciy commented on styles/styles.css:
```diff
@@ -39,6 +40,199 @@ body {
+}
+
+.topsection__date {
+    border: 3px solid #f5bd41;
```
Not sure it should be styled like active input by default.

5tronciy commented on styles/styles.css:
```diff
@@ -39,6 +40,199 @@ body {
+
+.topsection__inputs {
+    display: flex;
+    flex-direction: row;
```
You may miss default values.

5tronciy commented on styles/styles.css:
```diff
@@ -39,6 +40,199 @@ body {
+.topsection__inputs {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
```
It doesn't matter as container have no any gaps.

5tronciy commented on index.html:
```diff
@@ -10,24 +10,59 @@
+      <form class="topsection__form" action="./" method="get">
+        <label class="input__label topsection__input__label" for="city">Your destination or hotel name</label>
+        <label class="input__label topsection__input__label" for="date">Check-in &ndash; Check-out</label>
+        <label class="input__label topsection__input__label" for="guests"></label>
```
Label without content doesn't make sense for me.
Also, as the labels are separated from inputs - they appears in different way with inputs on screens with different width.

5tronciy reviewed (changes requested):
You need to improve. 
Also, I would call attention to the requirement of min-height in the task.

petrovsky747 committed:
Fixed layout of header block and form.
The "height" property has been added to the block "topsection", which sets its minimum height equal to the height of the screen.

5tronciy commented on index.html:
```diff
@@ -26,14 +26,18 @@
-        <label class="input__label topsection__input__label" for="date">Check-in &ndash; Check-out</label>
-        <label class="input__label topsection__input__label" for="guests"></label>
         <div class="topsection__inputs">
+          <label class="input__label topsection__input__label" for="city"></label>
```
Could you explain the sense of the label without content?
Read this [topic about `<label>` on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) to understand how we should use it.

5tronciy commented on styles/styles.css:
```diff
@@ -40,14 +39,17 @@ body {
 
+.topsection {
+    min-height: 100vh;
+}
```
Formally you have done the requirement. But there is an issue with fixing-sized the background. 
Try to repeat `min-height: 100vh;` to the block with the background image.

5tronciy commented on styles/styles.css:
```diff
@@ -155,17 +154,22 @@ body {
+.topsection__input__title {
+    padding-left: 104px;
+    display: flex;
+    gap: 220px;
```
It is not the best choice too. The labels should depend on the position of their input fields.

5tronciy reviewed (changes requested):
Research the using of tag label more pls. The layout needs improvement.

petrovsky747 committed:
Еach label of inputs is tied to its own input.
Reconfigured the display of the "topsection" block in the height of the screen.

5tronciy commented on styles/styles.css:
```diff
@@ -39,6 +39,211 @@ body {
+.topsection__city {
+    position: relative;
+    flex: auto;
+    padding-left: 22px;
```
Move this padding into the input if you want it to be highlighted properly if active.

5tronciy reviewed (approved):
Looks good!
